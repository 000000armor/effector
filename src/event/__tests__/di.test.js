//@flow

import {createEvent, type Event} from '..'

test('event.create single argument', () => {
  const foo = createEvent<number>('foo')
  const oldCreate = foo.create
  foo.create = jest.fn((payload, fullName, args) =>
    oldCreate(payload, fullName, args),
  )
  const baz = jest.fn()
  foo.watch(baz)
  foo(100)
  foo(200)
  foo(300)
  expect(baz.mock.calls).toMatchSnapshot()
  expect(foo.create.mock.calls).toMatchSnapshot()
})

type FnEvent<Fn> = Fn & Event<any>

test('event.create multiple arguments', () => {
  const bar: FnEvent<(number, string) => number> = createEvent<number>('bar')
  const oldCreate = bar.create
  bar.create = jest.fn((payload, fullName, args) =>
    oldCreate([payload, ...args], fullName, []),
  )
  const baz = jest.fn(payload => {})
  bar.watch(baz)
  bar(-2, 'foo')
  bar(-3, 'bar')
  bar(-2, 'baz')
  expect(baz.mock.calls).toMatchSnapshot()
  expect(bar.create.mock.calls).toMatchSnapshot()
})
