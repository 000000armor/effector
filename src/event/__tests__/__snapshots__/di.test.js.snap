// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event.create multiple arguments 1`] = `
Array [
  Array [
    Array [
      -2,
      "foo",
    ],
    "bar",
  ],
  Array [
    Array [
      -3,
      "bar",
    ],
    "bar",
  ],
  Array [
    Array [
      -2,
      "baz",
    ],
    "bar",
  ],
]
`;

exports[`event.create multiple arguments 2`] = `
Array [
  Array [
    -2,
    "bar",
    Array [
      "foo",
    ],
  ],
  Array [
    -3,
    "bar",
    Array [
      "bar",
    ],
  ],
  Array [
    -2,
    "bar",
    Array [
      "baz",
    ],
  ],
]
`;

exports[`event.create single argument 1`] = `
Array [
  Array [
    100,
    "foo",
  ],
  Array [
    200,
    "foo",
  ],
  Array [
    300,
    "foo",
  ],
]
`;

exports[`event.create single argument 2`] = `
Array [
  Array [
    100,
    "foo",
    Array [],
  ],
  Array [
    200,
    "foo",
    Array [],
  ],
  Array [
    300,
    "foo",
    Array [],
  ],
]
`;
